### PR TITLE
Make less known wayland compositors usable

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

They are currently either broken or not usable out of box. This PR fixes that.

way-cooler:
- [x] update to fix build https://github.com/NixOS/nixpkgs/pull/30088
- [x] fix crash
- [x] locker by alt+shift+l

via module
- [x] wc-grab and way-cooler-bg extensions
- [x] terminal by alt+enter
- [x] menu by alt+d
- [x] optional bar

orbment:
- [x] terminal by win+enter
- [x] menu by win+p

velox:
- [x] update to make it work
must be launched via swc-launch https://github.com/michaelforney/velox/issues/25
- [x] terminal by win+shift+enter
- [x] menu by win+r

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by peterhoeg** [comment]
Cc: @miltador as way-cooler maintainer.

**Comment by miltador** [comment]
LGTM for `way-cooler` changes.

**Comment by gnidorah** [comment]
Closing for one more change

**Comment by gnidorah** [comment]
I'm also interested in adding way-cooler module to this PR, if its author will provide lemonbar config

File: pkgs/applications/window-managers/orbment/default.nix [hunk 1]
```diff
@@ -1,9 +1,15 @@
-{ lib, stdenv, fetchgit, cmake, pkgconfig
+{ lib, stdenv, fetchgit, cmake, pkgconfig, makeWrapper, newScope
 , wlc, dbus_libs, wayland, libxkbcommon, pixman, libinput, udev, zlib, libpng
 , libdrm, libX11
+, westonLite
 }:
 
-stdenv.mkDerivation rec {
+let
+  callPackage =  newScope self;
```
Comment (FRidh) [inline pkgs/applications/window-managers/orbment/default.nix]: why not pass in `callPackage` instead?

Comment (gnidorah) [reply pkgs/applications/window-managers/orbment/default.nix]: @FRidh Fixed, thanks!

**Comment by gnidorah** [comment]
Ok, done.

File: pkgs/applications/window-managers/velox/default.nix [hunk 2]
```diff
@@ -1,25 +1,53 @@
...
+let
+  callPackage = newScope self;
+  self = {
+    swc = callPackage ./swc.nix {};
+    wld = callPackage ./wld.nix {};
+    dmenu-wayland = callPackage ./dmenu.nix {};
+    st-wayland = callPackage ./st.nix {
+      conf = stConf;
+      patches = stPatches;
+    };
```
Comment (zimbatm) [inline pkgs/applications/window-managers/velox/default.nix]: why not add these to all-packages.nix?

Comment (gnidorah) [reply pkgs/applications/window-managers/velox/default.nix]: @zimbatm As noted in https://github.com/NixOS/nixpkgs/pull/32285/commits/0520be305a95f70d92b312a7f3a06daf5b51363b these are unusable outside of velox.

**Comment by miltador** [comment]
Is it going to be merged?

**Comment by zimbatm** [comment]
Started nox-review on my build machine. If it passes I will merge.

**Comment by gnidorah** [comment]
@miltador @zimbatm Thanks!

**Pull request merged**